### PR TITLE
Final blog update for Belle II project

### PR DESCRIPTION
**Pull request by alexverus**
Update includes progress made post mid term, along with some concluding sections, screen-grabs and links to pull requests.

### CODE REVIEW

**Comment by alexverus** [comment]
Requesting for review from @hegner @agheata
My supervisor (@tkuhr) was unable to post comments here

**Review by agheata — CHANGES_REQUESTED** [review]
@alexverus the branch needs to be rebased on the current master. Please ask your mentor to confirm that he is OK with your blog (if unable to comment in the PR, just by mail to the admin list)

**Comment by tkuhr** [comment]
The blog correctly describes Arul's achievements. I approve it.

**Comment by hegner** [comment]
@tkuhr - thanks for the quick reaction.

**Comment by alexverus** [comment]
@agheata from what I can see, my branch doesn't seem to be behind upstream:master, apart from these commits. Could you kindly help me verify this?

**Comment by agheata** [comment]
> @agheata from what I can see, my branch doesn't seem to be behind upstream:master, apart from these commits. Could you kindly help me verify this?

If I do `gh pr checkout 1281` after fetching the upstream and rebasing my master on upstream/master, I get an error:
```
From https://github.com/HSF/hsf.github.io
 ! [rejected]        refs/pull/1281/head -> alexverus/master  (non-fast-forward)
exit status 1
```
Try to do on your local master:
```
git fetch upstream
git rebase upstream/master
git push [-f] # use -f if a simple push fails and proposes you a pull first
```

**Comment by alexverus** [comment]
> Try to do on your local master:

I had tried this earlier, and tried it again now and received `Everything up-to-date`. When I attempt `git diff master..upstream/master` I can see only one file changed(my blog post), and also on the web I see that there are no merge conflicts:
merge

I found that there's an open [issue](https://github.com/cli/cli/issues/562) with `gh pr checkout`, could it be possibly related to that? I could be doing something wrong too, haven't yet found out what though. Should I just close this and open a new request?

Apologies for the inconvenience.

File: _gsocblogs/2022/blog_Belle2_ArulPrakash.md [hunk 1]
```diff
@@ -25,19 +26,60 @@ These are the things that can be improved upon, in the existing system.
...
+ 
+ Whilst the base was being built on the sideline the existing implementation review was underway. The first aim was to get a birds-eye view of the overall system and then do a deep dive into the relevant modules, and identify the modules of interest. Time was also spent familiarizing myself with the software development and coding conventions. 
 
- #### CherrPy meets Gitlab
- Two of the key tech that will feature throughout the work are CherryPy and Gitlab, so a good amount of time was spent understanding and then getting hands dirty with trying and testing the integration of the two, which is key for the bulk of the improvements planned. With that, the first task undertaken was to extend details displayed along with plots of validation runs to include a list of relevant issues. A couple of features have also been added that would allow the reviewer of the results to directly create issues from the plot display window or update existing relevant issues in Gitlab. 
+ #### CherrPy meets GitLab
```
Comment (agheata) [inline _gsocblogs/2022/blog_Belle2_ArulPrakash.md:39]:
```suggestion
#### CherrPy meets GitLab
```

File: _gsocblogs/2022/blog_Belle2_ArulPrakash.md [hunk 2]
```diff
@@ -25,19 +26,60 @@ These are the things that can be improved upon, in the existing system.
...
+ 
+ Two of the key tech that will feature throughout the work are CherryPy and GitLab, so a good amount of time was spent understanding and then getting hands dirty with trying and testing the integration of the two, which is key for the bulk of the improvements planned. With that, the first task undertaken was to extend details displayed along with plots of validation runs to include a list of relevant issues. A couple of features have also been added that would allow the reviewer of the results to directly create issues from the plot display window or update existing relevant issues in GitLab. 
+ 
+<img src="https://github.com/alexverus/GSoC2022/blob/3d3032e945dac8281c7b342f8b74bb9864a7bc73/Plot.PNG?raw=true" alt="linked_issue_update" width = "100%">
+ 
+ #### Plot/logfile endpoints
```
Comment (agheata) [inline _gsocblogs/2022/blog_Belle2_ArulPrakash.md:45]:
```suggestion
#### Plot/logfile endpoints
```

File: _gsocblogs/2022/blog_Belle2_ArulPrakash.md [hunk 3]
```diff
@@ -25,19 +26,60 @@ These are the things that can be improved upon, in the existing system.
...
+ 
+ As a part of validation run result processing, emails are sent out for failed scripts/plots to the respective module owners. Apart from a brief summary regarding the plot/script responsible for the future, only links to the homepage of the validation was provided, forcing people to scour through the results to find the image of the failed plot/log file of the failed script. Modifications made to the mail-utils now will allow the mail-bot to include direct links to the relevant plots/script log files making it easier for the module owners to inspect and analyse what went wrong. 
+ 
+ #### Consolidate and display datafiles
```
Comment (agheata) [inline _gsocblogs/2022/blog_Belle2_ArulPrakash.md:49]:
```suggestion
#### Consolidate and display datafiles
```

File: _gsocblogs/2022/blog_Belle2_ArulPrakash.md [hunk 4]
```diff
@@ -25,19 +26,60 @@ These are the things that can be improved upon, in the existing system.
...
 
-### Journey so far
+### Project Progression
 
  #### Local base
- Initial days were spent setting up the basf2 software in the local system to be able to run the validation tool, host a local validation server and as well as to develop the code. Relevant data files were also obtained that are necessary inputs for certain validation modules. Finally, a Gitlab project was set up to mimic the production project and to be a playing field to test initial implementations. 
+ 
+ Initial days were spent setting up the basf2 software in the local system to be able to run the validation tool, host a local validation server and as well as to develop the code. Relevant data files were also obtained that are necessary inputs for certain validation modules. Finally, a GitLab project was set up to mimic the production project and to be a playing field to test initial implementations. 
 
  #### Code review
```
Comment (agheata) [inline _gsocblogs/2022/blog_Belle2_ArulPrakash.md:35]:
```suggestion
#### Code review 
```

File: _gsocblogs/2022/blog_Belle2_ArulPrakash.md [hunk 5]
```diff
@@ -25,19 +26,60 @@ These are the things that can be improved upon, in the existing system.
  * Provide developers with more information about failures found to help with root cause analysis. 
 
 
-### Journey so far
+### Project Progression
 
  #### Local base
```
Comment (agheata) [inline _gsocblogs/2022/blog_Belle2_ArulPrakash.md:31]:
```suggestion
#### Local base
```

**Review by agheata — COMMENTED** [review]
You were right, there is a bug in gh. There are just few subsection #### indenting problems, fixing them via the web interface.

**Review by agheata — APPROVED** [review]

**Comment by alexverus** [comment]
Thanks for the quick review and support @agheata @hegner @tkuhr

**Pull request merged**